test/tools/isoltest.cpp

```diff
@@ -480,9 +481,20 @@ int main(int argc, char const *argv[])
 
 		return global_stats ? 0 : 1;
 	}
-	catch (std::exception const& _exception)
+	catch (boost::program_options::error const& exception)
```

Comment:
I want to catch specifically `program_options::error` because I known it can be thrown in case of a command-line validation error. I do not expect `logic_error` here so if I get one I consider it a bug, to be handled the same way as an exception that should have been handled earlier and wasn't or an assertion failure.

To be precise, this specific catch branch could be placed just around the `options->parse()` call because that's the only place where we expect a CLI validation error from `boost::program_options` but I thought it would be better to place it next to the handler for `runtime_error`. They should really be handled the same way (print a nice message + exit).

And there are no other catches needed. I don't think there are any other types that can reach this handler in a normal flow and all unexpected ones and assertions are handled by `catch (...)`.